service/matching/tasklist/matcher.go

```diff
@@ -429,7 +429,14 @@ func (tm *taskMatcherImpl) Poll(ctx context.Context, isolationGroup string) (*In
 		TaskListKind: tm.tasklistKind.Ptr(),
 		EventName:    "Matcher Falling Back to Non-Local Polling",
 	})
-	return tm.pollOrForward(ctxWithCancelPropagation, startT, isolationGroup, isolatedTaskC, tm.taskC, tm.queryTaskC)
+	task, err := tm.pollOrForward(ctxWithCancelPropagation, startT, isolationGroup, isolatedTaskC, tm.taskC, tm.queryTaskC)
+	if task != nil {
```

Comment:
what about pollNonBlocking?

Reply:
+1

Reply:
I missed it after switching the places to record timer, thanks for catching it!

Reply:
Right, it is better to do it in a defer func to catch all return paths, updated